src/main/java/de/tum/in/www1/artemis/service/DataExportService.java

```diff
@@ -0,0 +1,565 @@
+
+import javax.annotation.Nullable;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.metis.AnswerPost;
+import de.tum.in.www1.artemis.domain.metis.Post;
+import de.tum.in.www1.artemis.domain.metis.Reaction;
+import de.tum.in.www1.artemis.domain.modeling.ModelingExercise;
+import de.tum.in.www1.artemis.domain.modeling.ModelingSubmission;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseStudentParticipation;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.domain.quiz.*;
+import de.tum.in.www1.artemis.repository.*;
+import de.tum.in.www1.artemis.repository.metis.AnswerPostRepository;
+import de.tum.in.www1.artemis.repository.metis.PostRepository;
+import de.tum.in.www1.artemis.repository.metis.ReactionRepository;
+import de.tum.in.www1.artemis.service.connectors.apollon.ApollonConversionService;
+import de.tum.in.www1.artemis.service.programming.ProgrammingExerciseExportService;
+import de.tum.in.www1.artemis.web.rest.dto.RepositoryExportOptionsDTO;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+
+/**
+ * Service Implementation for managing the data export in accordance with Art. 15 GDPR.
+ */
+@Service
+public class DataExportService {
+
+    private static final String ZIP_FILE_EXTENSION = ".zip";
+
+    private static final String CSV_FILE_EXTENSION = ".csv";
+
+    private static final String PDF_FILE_EXTENSION = ".pdf";
+
+    private static final String TXT_FILE_EXTENSION = ".txt";
+
+    private final Logger log = LoggerFactory.getLogger(DataExportService.class);
+
+    @Value("${artemis.data-export-path:./data-exports}")
+    private Path dataExportPath;
+
+    @Value("${artemis.repo-download-clone-path}")
+    private Path repoClonePath;
+
+    private final CourseRepository courseRepository;
+
+    private final UserRepository userRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    private final ZipFileService zipFileService;
+
+    private final ProgrammingExerciseExportService programmingExerciseExportService;
+
+    private final DataExportRepository dataExportRepository;
+
+    private final QuizQuestionRepository quizQuestionRepository;
+
+    private final QuizSubmissionRepository quizSubmissionRepository;
+
+    private final ExerciseRepository exerciseRepository;
+
+    private final DragAndDropQuizAnswerConversionService dragAndDropQuizAnswerConversionService;
+
+    // nullable in the constructor because otherwise the application doesn't start if the apollon profile is not set
+    private final ApollonConversionService apollonConversionService;
```

Comment:
In other parts of the code we use `Optional` in the constructor. Still not very pretty but at least you are forced to think about the presence of this service.